src/main/java/com/rocana/lucene/codec/v1/RocanaSearchCodecV1.java

```diff
@@ -32,15 +36,68 @@
  * the entire file improves Searcher open performance.
  *
  * If you have access to Rocana's JIRA, see ROCANA-8229.
+ *
+ * The idea behind this:
+ * 
+ * Ultimately we're trying to comment out one line of code in
+ * {@link RocanaBlockTreeTermsReader}'s constructor.
+ * 
+ * We don't want to customize Lucene's on-disk format. We just want
+ * to stop Lucene from checksumming an entire file when there's no
+ * benefit. The Lucene checksum is redundant since we store indexes
+ * on HDFS, which also checksums. The Lucene checksum happens at an
+ * inopportune time as we're trying to open a Rocana Search Searcher,
+ * which may happen during a rebalance, and we'd rather eliminate
+ * that penalty altogether or at least delay it until the rebalance
+ * finishes.
+ * 
+ * To accomplish that we register this class as the default Lucene
```

Comment:
I don't think we "register as the default". If we did the class path overriding then we would. Lucene has a method `Codec.getDefault()` and that will still return Lucene54Codec (I think - double check me on that). Instead, we "register an alternative Codec through the Lucene Codec SPI system" is probably how I would phrase it.